app/routers/using_shopping_list.py

```diff
@@ -0,0 +1,103 @@
+from datetime import datetime
+
+from fastapi import Depends, APIRouter, Form
+from fastapi.encoders import jsonable_encoder
+from requests import Session
+from sqlalchemy.exc import SQLAlchemyError
+from starlette import status
+from starlette.responses import RedirectResponse, JSONResponse
+
+from app.config import templates
+from app.dependencies import get_db
+from app.database.models import User
+from app.internal.create_shopping_list import create_shopping_product, by_id
+
+router = APIRouter(
+    prefix="/shopping_product",
+    tags=["shopping_product"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.post("/delete")
+def delete_shopping_product(
+        shopping_product_id: int = Form(...),
+        db: Session = Depends(get_db)
+):
+    # TODO: Check if the user is the owner of the shopping product.
+    shopping_product = by_id(db, shopping_product_id)
```

Comment:
Since you use `.one()` in `by_id`, please consider surrounding code with try/catch or change to `.first`.